tests/test_configdir.py

```diff
@@ -1,5 +1,150 @@
+        "INTERPOLATED_JSON": {
+            "redis": "redis://:$3cr3t@redis-sentinel:6319/mymaster/0",
+            "redis_master": ["mymaster"],
+        },
+        "INTERPOLATED_YAML": {
+            "redis": "redis://:$3cr3t@redis-sentinel:6319/mymaster/0"
+        },
+        "KEYS": {"keys": [{"id": 1, "private_key": "<private key pem>"}]},
+        "POSTGRES_DB_URI": "postgresql://root:postgres@postgres:5432/database",
+        "PUBLIC_KEY": b"<public key binary>",
+        "REDIS_HOST": "redis-sentinel",
+        "REDIS_PASSWORD": "$3cr3t",
+        "REDIS_PORT": "6319",
+        "REDIS_SENTINEL_MASTER": "mymaster",
+        "REDIS_URI": "redis://:$3cr3t@redis-sentinel:6319/mymaster/0",
+        "SMTP": {
+            "certificate": b"<binary certificate>",
+            "host": "mail.mailgun.com",
+            "password": "$3cr3t",
+            "port": "542",
+            "username": "sender",
+        },
+        "UNICODE": "∫å∂¬∑π∑",
+    }
+
+
+def test_variable_interpolation():
+    """
+    Verify variable interpolation placeholders are replaced with the referenced value.
+    """
+    values = {
+        "UNICODE": "∫å∂¬∑π∑",
+        "SMTP": {
+            "certificate": b"<binary>",
+            "host": "mail.mailgun.com",
+            "password": "$3cr3t",
+            "port": "542",
+            "username": "sender",
+        },
+        "KEYS": {"keys": [{"id": 1, "private_key": "<private key pem>"}]},
+        "other_settings": {
+            "host": "{{SMTP.host}}",
+            "user_and_password": ["{{SMTP.username}}", "{{SMTP.password}}"],
+        },
+    }
+    interpolate = Interpolator(values)
+
+    assert interpolate("{{UNICODE}}") == "∫å∂¬∑π∑"
+    assert interpolate("{{SMTP.host}}") == "mail.mailgun.com"
+    assert interpolate("{{KEYS.keys@0.private_key}}") == "<private key pem>"
+
+    # Referencing an integer is OK, but it comes out as a string
+    assert interpolate("{{KEYS.keys@0.id}}") == "1"
+
+    # Can not reference binary
+    with pytest.raises(TypeError):
+        interpolate("{{SMTP.certificate}}")
+    # Can not reference a dictionary
+    with pytest.raises(TypeError):
+        interpolate("{{KEYS}}")
+    # Can not reference a list
+    with pytest.raises(TypeError):
+        interpolate("{{KEYS.keys}}")
+    # Invalid list index
+    with pytest.raises(IndexError):
+        interpolate("{{KEYS.keys@1.private_key}}")
+    # Invalid dictionary key
+    with pytest.raises(KeyError):
+        interpolate("{{KEYS.not_a_key@0.private_key}}")
+
+
+def test_config_dir_directory(monkeypatch):
+    """
+    Verify `configdir` directory:
+
+    - Defaults to /configdir
+    - Can be overridden with env var
+    """
+    def _just_return_dir(directory):
+        return directory
+    monkeypatch.setattr("configdir.parse", _just_return_dir)
+
+    assert configdir() == "/configdir"
+    assert configdir("/foo") == "/foo"
+
+    monkeypatch.setitem(os.environ, "CONFIGDIR", "/bar")
```

Comment:
monkeypatch.setenv("CONFIGDIR", "/bar")

Reply:
That doesn't work with py2 and 3. In py2 it complains that values must be bytes. In py3 it complains values must be str. I didn't want to have a `if PY3` switch.